src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs

```diff
@@ -183,13 +183,19 @@ private async Task ShowToolWindowAsync(CancellationToken cancellationToken)
                 return null;
             }
 
+            RoslynDebug.AssertNotNull(roslynPackage.SolutionEventMonitor);
+
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             if (ValueTrackingToolWindow.Instance is null)
             {
                 var factory = roslynPackage.GetAsyncToolWindowFactory(Guids.ValueTrackingToolWindowId);
 
-                var viewModel = new ValueTrackingTreeViewModel(_classificationFormatMapService.GetClassificationFormatMap(textView), _typeMap, _formatMapService);
+                var viewModel = new ValueTrackingTreeViewModel(
+                    _classificationFormatMapService.GetClassificationFormatMap(textView),
+                    _typeMap,
+                    _formatMapService,
+                    roslynPackage.SolutionEventMonitor);
```

Comment:
do you have a workspace?  if so, you can listen for SolutionCleared/SolutionRemoved events.

Reply:
i would personally prefer that to avoid taking this strong dependency on the package and this specific vs detail.

Reply:
I do have a workspace. I knew there was solution changed, did not know we had solution cleared/removed events. Why do we have those? Doesn't the workspace get cleaned up if we have no solution?